### ISSUE TITLE
Sample/simulate from a EpiProblem component

### ISSUE THREAD

**Issue by seabbs** [CDCgov/Rt-without-renewal#323]
At the moment we show how to construct `EpiProblems` from individual submodels and we show how these can be simulated from/sampled individually. Do we want to think about functionality so that you can do this directly on a EpiProblem object and potentially on parts of the EpiProblem composed model?

This is a bit of a late night issue so will circle back with more thoughts soon.

**Comment by SamuelBrand1** [comment]
The `DirectSample` `EpiMethod` does this? What would you like to add?

**Comment by seabbs** [comment]
It allows you to sample from the whole object. I am talking about just sampling from say the infection or observation model. I think it would just be a EpiProblem method for `generate_infections` and `generate_observations`?

**Comment by SamuelBrand1** [comment]
Right, I misunderstood. A method that does this would be v straightforward.
